pkg/diskrepart/mkfs.go

```diff
@@ -60,6 +85,10 @@ func (mkfs MkfsCall) buildOptions() ([]string, error) {
 			opts = append(opts, "-n")
 			opts = append(opts, mkfs.label)
 		}
+		if mkfs.uuid != "" {
+			opts = append(opts, "-i")
+			opts = append(opts, strings.Split(mkfs.uuid, "-")[0])
```

Comment:
VFat filesyste does not really have UUIDs, however it is a common practice to use the VolumeID as such. The VFat volume ID is represented by 8 hex values, like the first chunk of UUID bytes, so for ease of use, to define the vfat volume ID a valid UUID is required and only the first 8 bytes will be used to defined the volumeID.